packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -437,7 +438,7 @@ export const useGridFilter = (
         filteredDescendantCountLookup: {},
       };
     },
-    [apiRef, props.filterMode, rows],
+    [apiRef, props, rows],
```

Comment:
Wouldn't this save some unnecessary recomputations? (Keeping in mind that `getRowId` has a stable reference)
```suggestion
    [apiRef, props.filterMode, props.getRowId, rows],
```

Reply:
The linter wanted `props` in the dependencies array for some reason. I've refactored to include those fields.